sbor/src/payload_validation/payload_validator.rs

```diff
@@ -182,11 +185,33 @@ pub fn validate_container<E: CustomTypeExtension>(
     Ok(())
 }
 
-pub fn validate_terminal_value<'de, E: CustomTypeExtension>(
+pub fn validate_terminal_value<'de, E: ValidatableCustomTypeExtension<T>, T>(
     schema: &Schema<E>,
     value: &TerminalValueRef<'de, E::CustomTraversal>,
     type_index: LocalTypeIndex,
+    context: &mut T,
```

Comment:
I'm surprised the context needs to be mut, but I guess it's okay... did you consider using interior mutability instead?

Reply:
I thought about that, and even started with `&T`.  It requires a much larger refactoring to the engine, so gave up.

Reply:
Well you just create:
```
ScryptoValidationContext { system_service: RefCell<&'_ mut SystemService> }
```

And then do `system_service = context.system_service.borrow_mut()` :)

Reply:
Although that can't implement Copy if that's a problem :')